test/compress/harmony.js

```diff
@@ -444,3 +476,66 @@ generators_yield_assign: {
     }
     expect_exact: "function*fn(){var x={};x.prop=yield 5}"
 }
+
+generator_yield_undefined: {
+    input: {
+        function* fn() {
+            yield;
+        }
```

Comment:
Yeah, this would be interesting to see if undefined could be dropped by compression or without. Especially with yield* as removing undefined might make the code invalid.